Insights/Host/HostIPStats.yaml

```diff
@@ -0,0 +1,349 @@
+  | extend Direction = iff(CommunicationDirection !in~ ('outbound','0') or CommunicationDirection !in~ ('inbound','1'), 'NotAvailable', CommunicationDirection)
+  | summarize StartTime = min(TimeGenerated), EndTime = max(TimeGenerated), InboundCount = countif(Direction in~ ('Inbound','1')), OutboundCount = countif(Direction in~ ('Outbound', '0')), UnknownDirection = countif(Direction =~ 'NotAvailable') by HostIPAddress = SourceIP, Type = strcat(Type,':', DeviceVendor,'-', DeviceProduct), RemoteIPAddress = DestinationIP, Direction, SentBytes = tolong(SentBytes)
+  ),
+  (
+  CommonSecurityLog
+  | where TimeGenerated between (start_time..end_time)
+  | where DestinationIP in (IpList)
+  | where SourceIP != DestinationIP
+  //| where DeviceAction has_any ('allow', 'allowed', 'accept', 'built', 'start', 'connect', '')
+  //| where not(DeviceAction has_any ('built','deny', 'denied', 'rst', 'blocked', 'teardown'))
+  | extend Direction = iff(CommunicationDirection !in~ ('outbound','0') or CommunicationDirection !in~ ('inbound','1'), 'NotAvailable', CommunicationDirection)
+  | summarize StartTime = min(TimeGenerated), EndTime = max(TimeGenerated), InboundCount = countif(Direction in~ ('Inbound','1')), OutboundCount = countif(Direction in~ ('Outbound', '0')), UnknownDirection = countif(Direction =~ 'NotAvailable') by HostIPAddress = DestinationIP, Type = strcat(Type,':', DeviceVendor,'-', DeviceProduct), RemoteIPAddress = SourceIP, Direction, ReceivedBytes = tolong(ReceivedBytes)
+  )
+  );
+  IpStats | join kind=leftouter Host_Account on HostIPAddress | project-away HostIPAddress1
+  };
+  GetIPResolutionWithHost('{{Host_HostName}}', '{{Host_NTDomain}}', '{{Host_DnsDomain}}', '{{Host_AzureID}}', '{{Host_OMSAgentID}}')
+
+Insights:
+ Id: 2ed19a9a-7ebe-4941-a6a0-fa0b020c48da
+ DisplayName: "Host IP address remote connections"
+ Description: |
+  'Provides the remote IP connection count information for outbound and inbound connections to an IP address associated with a given host. Note due to potential performance impact, data is limited to a 7 day max window.'
+ DefaultTimeRange: 
+  BeforeRange: 12h
+  AfterRange: 12h
+ TableQuery:
+  ColumnsDefinitions:
+  - Header: "Direction"
+    OutputType: String
+    SupportDeepLink: false
+  - Header: "HostIP"
+    OutputType: String
+    SupportDeepLink: true
+  - Header: "RemoteIP/Count"
+    OutputType: String
+    SupportDeepLink: true
+  - Header: "Total"
+    OutputType: Number
+    SupportDeepLink: true
+  QueriesDefinitions:
+  # Top inbound counts
+  - Filter:    "where InboundCount > 0"
+    Summarize: "summarize Total = sum(InboundCount) by HostIPAddress, RemoteIPAddress | top 1 by Total"
+    Project:   "project Direction = 'Top In', HostIP = HostIPAddress, RemoteIP = RemoteIPAddress, Total"
+    LinkColumnsDefinitions:
+    - ProjectedName: Direction
+      Query: "{{BaseQuery}} | {{RowFilter}}"
+    - ProjectedName: HostIP
+      Query: "{{BaseQuery}} | {{RowFilter}}"
```

Comment:
These all link to the same query. Is this intentional? Wouldn't linking only the "Direction" column be enough?

Reply:
This is the default for each of the deeplinks, it is not required, but I include them so it is clear what the deeplink is using.